homeassistant/components/knx/climate.py

```diff
@@ -34,21 +37,50 @@ async def async_setup_platform(
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up climate(s) for KNX platform."""
+    await _migrate_unique_id(hass, discovery_info)
     entities = []
     for device in hass.data[DOMAIN].xknx.devices:
         if isinstance(device, XknxClimate):
             entities.append(KNXClimate(device))
     async_add_entities(entities)
 
 
+async def _migrate_unique_id(
+    hass: HomeAssistant, discovery_info: DiscoveryInfoType | None
+) -> None:
+    """Change unique_ids used in 2021.4 to include target_temperature GA."""
+    entity_registry = er.async_get(hass)
+    if not discovery_info or not discovery_info["platform_config"]:
+        return
+
+    platform_config = discovery_info["platform_config"]
+    for entity_config in platform_config:
+        # normalize group address strings - ga_temperature_state was the old uid
+        ga_temperature_state = parse_device_group_address(
+            entity_config[ClimateSchema.CONF_TEMPERATURE_ADDRESS][0]
+        )
+        old_uid = str(ga_temperature_state)
+
+        entity_id = entity_registry.async_get_entity_id("climate", DOMAIN, old_uid)
+        if entity_id is not None:
+            ga_target_temperature_state = parse_device_group_address(
+                entity_config[ClimateSchema.CONF_TARGET_TEMPERATURE_STATE_ADDRESS][0]
+            )
+            new_uid = f"{ga_temperature_state}_{ga_target_temperature_state}"
+            entity_registry.async_update_entity(entity_id, new_unique_id=new_uid)
```

Comment:
It's nice to prefer guard clause and keep indentation low.
```suggestion
        if entity_id is None:
            continue
        ga_target_temperature_state = parse_device_group_address(
            entity_config[ClimateSchema.CONF_TARGET_TEMPERATURE_STATE_ADDRESS][0]
        )
        new_uid = f"{ga_temperature_state}_{ga_target_temperature_state}"
        entity_registry.async_update_entity(entity_id, new_unique_id=new_uid)
```